File: packages/cli-js/src/utils/prompt.ts
/**
 * Load prompt content from a file path (LangChain-style load_prompt).
 * No fallback: file must exist; throws on read error or missing file.
 */

import { readFile } from "fs/promises";

/**
 * Load prompt text from a file. Returns trimmed content.
 * @param path - Absolute path to the prompt file (e.g. .md).
 * @throws If the file cannot be read or is missing.
 */
export async function loadPrompt(path: string): Promise<string> {
  const text = await readFile(path, "utf-8");
  return text.trim();
}

File: packages/cli-py/planforge/utils/prompt.py
"""Load prompt content from a file path (LangChain-style load_prompt). No fallback: file must exist."""

from pathlib import Path


def load_prompt(path: str | Path) -> str:
    """Load prompt text from a file. Returns stripped content. Raises if file is missing or unreadable."""
    p = Path(path)
    return p.read_text(encoding="utf-8").strip()
